Update Haskell package set to Stackage Nightly 2021-02-02 (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on Friday, 2021-02-05 20:00 +01:00. You can watch this live on Twitch at https://www.twitch.tv/peti343.

## TODO

- [x] https://github.com/NixOS/docker/pull/31: Use a pure Nix approach to building containers

TerrorJack committed:
Fix ar command path in GHC.

Previously, the "ar command" in the global config of GHC in nixpkgs is
simply "ar" instead of a proper absolute path in the nix store. This
will result in an "ar: command not found" error when using GHC and cabal
in a pure nix shell. This commit adds the patch and applies to all
pre-9.0 versions.

See output of ghc --info for "ar command" value.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-10-gabc66d1 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/d5c9f894f31fba43db8123a6776a4819b04ef3c4.

danielstaleiny committed:
haskellPackages.postgrest: broken -> unbroken
Tested on NixOS

iyefrat committed:
haskellPackages.hum: broken -> unbroken

cdepillabout committed:
spago: fix build by specifying older version of dhall

peti committed:
Stackage Nightly 2021-02-02

SuperSandro2000 commented on pkgs/development/compilers/ghc/8.10.3.nix:
```diff
@@ -107,7 +107,15 @@ stdenv.mkDerivation (rec {
+    # upstream patch. Don't forget to check backport status of the upstream patch
+    # when adding new GHC releases in nixpkgs.
+    ./respect-ar-path.patch
+  ] ++ stdenv.lib.optionals stdenv.isDarwin [
```
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
```

SuperSandro2000 commented:
@ofborg eval

peti committed:
ghc-8.10.3: don't use obsolete stdenv.lib

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

maralorn committed:
haskell-language-server: Use hackage version and cleanup

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-10-gabc66d1 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/5ad244e7f59d62ef878c4ca33dc18dc126d4a202.

peti committed:
hackage2nix: disable failing builds to fix evaluation on Hydra

peti committed:
essence-of-live-coding-warp: update overrides for the new version of http-client

maralorn committed:
haskell.ghc865.mmorph: Fix build

peti committed:
hls-hlint-plugin: add new ghc-lib dependency when building with older compilers

peti committed:
haskell-syb: drop obsolete overrides to fix the ghc-9.0.x build

peti committed:
ghc: update the 9.0.1 version to the official release tarball

maralorn commented:
@peti Have you already opened a new PR?

peti commented:
Now I have. Tonight, we'll merge https://github.com/NixOS/nixpkgs/pull/112893.
